### PR TITLE
Add step for Jetpack Cancellation flow to show plan/ product information and benefits

### PR DESCRIPTION
**Pull request by jboland88**
#### Changes proposed in this Pull Request

* This PR is related to/ a subpart of https://github.com/Automattic/wp-calypso/pull/54669 which introduces a new cancellation flow for Jetpack products and plans. 
* This PR adds the Benefits step component for the cancellation flow 

#### Testing instructions

The visual output of this component can be seen by testing PR #54669
* Either check out the branch from PR #54669 locally or use the Calypso live link on that branch to get a preview. 
* Review the testing instructions on PR #54669 to check the visual output

### CODE REVIEW

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 1]
```diff
@@ -0,0 +1,159 @@
...
+
+		return moment.duration( purchaseExpiryDate.diff( moment( purchase.mostRecentRenewDate ) ) );
+	};
+
+	const getTimeRemainingTranslatedPeriod = ( purchase: Purchase ) => {
+		const timeRemaining = getTimeRemainingForSubscription( purchase );
+
+		if ( timeRemaining.months() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.months();
+			const periodName = translate( 'month', 'months', { count: timeRemaining.months() } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'month', 'months', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 16*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 2]
```diff
@@ -0,0 +1,159 @@
...
+
+			return { timeRemainingNumber, periodName };
+		} else if ( timeRemaining.weeks() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.weeks();
+			const periodName = translate( 'week', 'weeks', { count: timeRemainingNumber } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'week', 'weeks', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 17*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+
+			return { timeRemainingNumber, periodName };
+		}
+
+		const timeRemainingNumber = timeRemaining.days();
+		const periodName = translate( 'day', 'days', { count: timeRemainingNumber } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'day', 'days', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx (hunk 1 above)
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: Nitpick: I think `timeRemaining.months()` on this line was intending to use the variable `timeRemainingNumber`

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 4]
```diff
@@ -0,0 +1,159 @@
...
+						},
+						comment:
+							"'period' is either one of 'day', 'week', 'month', or their plural form. 'timeRemaining' is a number representing the time left that will be used with the 'period'.",
+					}
+				) }
+			</React.Fragment>
+		);
+	};
+
+	const getCancelConsequenceByProduct = ( productSlug: string ) => {
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:119]: Not a dev task - just a note. Once we have this merged into the test branch, we can ask Rob/ Simon to review these strings. I just made these up 🙂

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 5]
```diff
@@ -0,0 +1,159 @@
...
+		if ( isJetpackScanSlug( productSlug ) ) {
+			return translate(
+				'Once you cancel your site will no longer have automatic protection from threats.'
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: Looks like I forgot to add a comma on all of these after "Once you cancel". Should read "Once you cancel, ..."

**Review by jboland88 — COMMENTED** [review]
A couple of tiny nitpicks, otherwise this one LGTM code-wise. I can test the output when this is merged into the main PR 👍 
Nice work on the translation handling.

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 6]
```diff
@@ -0,0 +1,159 @@
...
+
+		return moment.duration( purchaseExpiryDate.diff( moment( purchase.mostRecentRenewDate ) ) );
+	};
+
+	const getTimeRemainingTranslatedPeriod = ( purchase: Purchase ) => {
+		const timeRemaining = getTimeRemainingForSubscription( purchase );
+
+		if ( timeRemaining.months() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.months();
+			const unitOfTime = translate( 'month', 'months', { count: timeRemainingNumber } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'month', 'months', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 16*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 7]
```diff
@@ -0,0 +1,159 @@
...
+
+			return { timeRemainingNumber, unitOfTime };
+		} else if ( timeRemaining.weeks() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.weeks();
+			const unitOfTime = translate( 'week', 'weeks', { count: timeRemainingNumber } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'week', 'weeks', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 17*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 8]
```diff
@@ -0,0 +1,159 @@
...
+
+			return { timeRemainingNumber, unitOfTime };
+		}
+
+		const timeRemainingNumber = timeRemaining.days();
+		const unitOfTime = translate( 'day', 'days', { count: timeRemainingNumber } );
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'day', 'days', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 9]
```diff
@@ -0,0 +1,159 @@
...
+	};
+
+	const getCancelConsequenceByProduct = ( productSlug: string ) => {
+		if ( isJetpackScanSlug( productSlug ) ) {
+			return translate(
+				'Once you cancel, your site will no longer have automatic protection from threats.'
+			);
+		} else if ( isJetpackBackupSlug( productSlug ) ) {
+			return translate( 'Once you cancel, you will lose access to your site backups.' );
+		} else if ( isJetpackSearch( productSlug ) ) {
```
Comment (jboland88) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: @IanRamosC  - In doing some testing - I remembered that `isJetpackSearch` only takes a product object and not a slug string. Alternatively, I have used `JETPACK_SEARCH_PRODUCTS.includes()` elsewhere to match against the array of search product slugs

Comment (IanRamosC) [reply client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx]: Nice catch! Fixed the usage of that here.

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 10]
```diff
@@ -0,0 +1,168 @@
...
+
+		return moment.duration( purchaseExpiryDate.diff( moment( purchase.mostRecentRenewDate ) ) );
+	};
+
+	const getTimeRemainingTranslatedPeriod = ( purchase: Purchase ) => {
+		const timeRemaining = getTimeRemainingForSubscription( purchase );
+
+		if ( timeRemaining.months() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.months();
+			const unitOfTime = translate( 'month', 'months', {
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:43]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'month', 'months', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)


:information_source: This string already exists with the following **contexts**:
- `Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.`

 Would it make sense to reuse one of the above?

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 11]
```diff
@@ -0,0 +1,168 @@
...
+				count: timeRemainingNumber,
+				context: 'unit of time',
+			} );
+
+			return { timeRemainingNumber, unitOfTime };
+		} else if ( timeRemaining.weeks() >= 1 ) {
+			const timeRemainingNumber = timeRemaining.weeks();
+			const unitOfTime = translate( 'week', 'weeks', {
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:51]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **22** times:
`translate( 'week', 'weeks', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 14*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)


:information_source: This string already exists with the following **contexts**:
- `Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.`

 Would it make sense to reuse one of the above?

File: client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx [hunk 12]
```diff
@@ -0,0 +1,168 @@
...
+				count: timeRemainingNumber,
+				context: 'unit of time',
+			} );
+
+			return { timeRemainingNumber, unitOfTime };
+		}
+
+		const timeRemainingNumber = timeRemaining.days();
+		const unitOfTime = translate( 'day', 'days', {
```
Comment (a8ci18n) [inline client/components/marketing-survey/cancel-jetpack-form/jetpack-benefits-step.tsx:60]: :information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **23** times:
`translate( 'day', 'days', { context: 'Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.'} )` *ES Score: 12*
See 1 additional suggestions in the [PR translation status page](https://translate.wordpress.com/localci/status/automattic/wp-calypso/add/jetpack-cancellation-flow-benefits-step)


:information_source: This string already exists with the following **contexts**:
- `Used in the trial period dropdown. Number is in text field. 0, 2+ will need plural, 1 will need singular.`

 Would it make sense to reuse one of the above?

**Review by IanRamosC — APPROVED** [review]

**Comment by a8ci18n** [comment]
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/6455351

Hi @jboland88, could you please edit the description of this PR and add a screenshot for our translators? Ideally it'd include all of the following strings:
- `Your {{strong}} %(productName)s {{/strong}} subscription is still active. {{br/}}`
- `You still have {{strong}}%(timeRemaining)d %(unitOfTime)s{{/strong}} left on your {{strong}}%(productName)s{{/strong}} subscription. {{br/}}`
- `Once you cancel, your site will no longer have automatic protection from threats.`
- `Once you cancel, you will lose access to your site backups.`
- `Once you cancel, you will no longer have Jetpack's enhanced search experience.`
- `Once you cancel, you will lose access to the following:`
- `Additionally, you will lose access to:`
- `month`
- `week`
- `day`

Thank you in advance!

**Comment by a8ci18n** [comment]
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/6455351).

**Pull request merged**